bson-record-codec/src/main/org/bson/codecs/record/RecordCodec.java

```diff
@@ -156,6 +158,14 @@ private static String computeFieldName(final RecordComponent component) {
             return component.getName();
         }
 
+        private static boolean checkFieldIsNullable(final RecordComponent component) {
```

Comment:
This should be replaced with a more straightforward check of 

```
!component.getType().isPrimitive()
```

Not everyone uses the `javax.annotation.Nullable` type to declare nullable components.  If a component value shouldn't be null, the constructor can throw.

Reply:
Make sense - updated here, test record name and PR decription.